src/Files.App/Extensions/WebView2Extensions.cs

```diff
@@ -0,0 +1,222 @@
+﻿// Copyright (c) Mahmoud Al-Qudsi, NeoSmart Technoogies. All rights reserved.  
+// Licensed under the MIT License.
+
+using CommunityToolkit.WinUI;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.Web.WebView2.Core;
+using Newtonsoft.Json;
+using System.Reflection;
+using System.Text;
+using Windows.Foundation;
+
+namespace Files.App.Extensions
+{
+	/// <summary>
+	/// Code modified from https://gist.github.com/mqudsi/ceb4ecee76eb4c32238a438664783480
+	/// </summary>
+	public static class WebView2Extensions
+	{
+		public static void Navigate(this WebView2 webview, Uri url)
+		{
+			webview.Source = url;
+		}
+
+		private enum PropertyAction
+		{
+			Read = 0,
+			Write = 1,
+		}
+
+		private struct WebMessage
+		{
+			public Guid Guid { get; set; }
+		}
+
+		private struct MethodWebMessage
+		{
+			public string Id { get; set; }
+			public string Method { get; set; }
+			public string Args { get; set; }
+		}
+
+		private struct PropertyWebMessage
+		{
+			public string Id { get; set; }
+			public string Property { get; set; }
+			public PropertyAction Action { get; set; }
+			public string Value { get; set; }
+		}
+
+		public static List<TypedEventHandler<WebView2, CoreWebView2WebMessageReceivedEventArgs>> _handlers = new List<TypedEventHandler<WebView2, CoreWebView2WebMessageReceivedEventArgs>>();
```

Comment:
You probably can use a [type alias](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-12.0/using-alias-types) for this `List<TypedEventHandler<WebView2, CoreWebView2WebMessageReceivedEventArgs>>` big guy.